develop/buf-breaking.sh

```diff
@@ -61,26 +64,24 @@ check_against_commit() {
   fi
 }
 
-if [[ $PR_BASE_COMMIT ]]; then
+if [[ $PR_BASE_COMMIT != $COMMIT ]]; then
```

Comment:
Is the intention here to just avoid running the check if it's going to diff against the same thing? Note that it's not diffing HEAD vs PR_BASE_COMMIT, it's diffing HEAD vs merge-base(HEAD, PR_BASE_COMMIT). But I guess it has the same effect (unless PR_BASE_COMMIT is something on main behind the head of main, which can't happen).

Anyway, I was using the presence of PR_BASE_COMMIT as a signal that we're running in CI. If we're not in CI, I wanted to avoid doing `git fetch` on this repo (the user's main repo) and wanted to do it on the clone only.

There's not much harm in doing a git fetch but I like the property that this script doesn't modify the repo that it's run in (except in CI).

At the very least, the comment here is wrong now.

Reply:
This is for the case you're manually triggering the GHA to run tests on a commit already merged to main branch. In this case, the `PR_BASE_COMMIT` is the same as `COMMIT`.